drivers/gpio/gpio_pca_series.c

```diff
@@ -0,0 +1,2178 @@
+						   enum gpio_pca_series_reg_type reg_type,
+						   uint8_t *buf);
+#endif /* CONFIG_GPIO_PCA_SERIES_CACHE_ALL */
+
+/**
+ * @brief read register with i2c interface.
+ *
+ * @note if CONFIG_GPIO_PCA_SERIES_CACHE_ALL is enabled, this will
+ *          also update reg cache.
+ *
+ * @param dev
+ * @param reg_type
+ * @param buf pointer to read data
+ * @return int 0        if success
+ *             -EFAULT  if register is not supported
+ *             -EIO     if i2c failure
+ */
+static inline int gpio_pca_series_reg_read(const struct device *dev,
+					   enum gpio_pca_series_reg_type reg_type, uint8_t *buf)
+{
+	const struct gpio_pca_series_config *cfg = dev->config;
+	int ret;
+	uint32_t size = gpio_pca_series_reg_get_size(dev, reg_type);
+	uint8_t addr = gpio_pca_series_reg_get_addr(dev, reg_type);
+	struct i2c_msg msg[2];
+
+	LOG_DBG("reg read type %d addr 0x%x len %d", reg_type, addr, size);
+
+	if (addr == PCA_REG_INVALID) {
+		LOG_ERR("trying to read unsupported reg, reg type %d", reg_type);
+		return -EFAULT;
+	}
+
+	msg[0].buf = &addr;
+	msg[0].len = 1;
+	msg[0].flags = I2C_MSG_WRITE;
+	msg[1].buf = buf;
+	msg[1].len = size;
+	msg[1].flags = I2C_MSG_RESTART | I2C_MSG_READ | I2C_MSG_STOP;
+
+	ret =  i2c_transfer(dev, msg, 2, addr);
+	if (ret) {
+		LOG_ERR("i2c read error [%d]", ret);
+		return ret;
+	}
+
+#ifdef CONFIG_GPIO_PCA_SERIES_CACHE_ALL
+	gpio_pca_series_reg_update_cache(dev, reg_type, buf);
+#endif /* CONFIG_GPIO_PCA_SERIES_CACHE_ALL */
+
+	LOG_DBG("reg read return %d", ret);
+	return ret;
+}
+
+/**
+ * @brief write register with i2c interface.
+ *
+ * @note if CONFIG_GPIO_PCA_SERIES_CACHE_ALL is enabled, this will
+ *          also update reg cache.
+ *
+ * @param dev
+ * @param reg_type
+ * @param buf pointer to read data
+ * @return int 0        if success
+ *             -EFAULT  if register is not supported
+ *             -EIO     if i2c failure
+ */
+static inline int gpio_pca_series_reg_write(const struct device *dev,
+					enum gpio_pca_series_reg_type reg_type, uint8_t *buf)
+{
+	const struct gpio_pca_series_config *cfg = dev->config;
+	int ret;
+	uint32_t size = gpio_pca_series_reg_get_size(dev, reg_type);
+	uint8_t addr = gpio_pca_series_reg_get_addr(dev, reg_type);
+	struct i2c_msg msg[2];
+
+	LOG_DBG("reg write type %d addr 0x%x len %d", reg_type, addr, size);
+
+	if (addr == PCA_REG_INVALID) {
+		LOG_ERR("trying to write unsupported reg");
+		return -EFAULT;
+	}
+
+	msg[0].buf = &addr;
+	msg[0].len = 1;
+	msg[0].flags = I2C_MSG_WRITE;
+	msg[1].buf = buf;
+	msg[1].len = size;
+	msg[1].flags = I2C_MSG_WRITE | I2C_MSG_STOP;
+
+	ret =  i2c_transfer(dev, msg, 2, addr);
+	if (ret) {
+		LOG_ERR("i2c write error [%d]", ret);
+		return ret;
+	}
+
+#ifdef CONFIG_GPIO_PCA_SERIES_CACHE_ALL
+	gpio_pca_series_reg_update_cache(dev, reg_type, buf);
+#endif /* CONFIG_GPIO_PCA_SERIES_CACHE_ALL */
+
+	LOG_DBG("reg write return %d", ret);
+	return ret;
+}
+
+/**
+ * }
+ * // gpio_pca_reg_access_api
```

Comment:
Nit- please drop ``//`` style comments within this file- I think the CI check is not catching them since they are part of a C89 style comment block, but they should still be removed.